rest/route/service.go

```diff
@@ -113,13 +114,13 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/tasks/{task_id}").Version(2).Get().Wrap(checkUser).RouteHandler(makeGetTaskRoute(sc))
 	app.AddRoute("/tasks/{task_id}").Version(2).Patch().Wrap(checkUser, addProject).RouteHandler(makeModifyTaskRoute(sc))
 	app.AddRoute("/tasks/{task_id}/abort").Version(2).Post().Wrap(checkUser).RouteHandler(makeTaskAbortHandler(sc))
-	app.AddRoute("/tasks/{task_id}/generate").Version(2).Post().RouteHandler(makeGenerateTasksHandler(sc, opts.QueueGroup))
-	app.AddRoute("/tasks/{task_id}/generate").Version(2).Get().RouteHandler(makeGenerateTasksPollHandler(sc, opts.QueueGroup))
+	app.AddRoute("/tasks/{task_id}/generate").Version(2).Post().Wrap(checkTask).RouteHandler(makeGenerateTasksHandler(sc, opts.QueueGroup))
+	app.AddRoute("/tasks/{task_id}/generate").Version(2).Get().Wrap(checkTask).RouteHandler(makeGenerateTasksPollHandler(sc, opts.QueueGroup))
 	app.AddRoute("/tasks/{task_id}/restart").Version(2).Post().Wrap(addProject, checkUser).RouteHandler(makeTaskRestartHandler(sc))
 	app.AddRoute("/tasks/{task_id}/tests").Version(2).Get().Wrap(addProject).RouteHandler(makeFetchTestsForTask(sc))
 	app.AddRoute("/user/settings").Version(2).Get().Wrap(checkUser).RouteHandler(makeFetchUserConfig())
 	app.AddRoute("/user/settings").Version(2).Post().Wrap(checkUser).RouteHandler(makeSetUserConfig(sc))
-	app.AddRoute("/user/author/{user_id}").Version(2).Get().Wrap(checkUser).RouteHandler(makeFetchUserAuthor(sc))
+	app.AddRoute("/user/author/{user_id}").Version(2).Get().Wrap(checkTask).RouteHandler(makeFetchUserAuthor(sc))
```

Comment:
Probably no one is using this, but it would still be good to handle authentication of (task or user). Since UIv2 will be using these routes, it will come in useful later. How difficult would it be to adjust the existing user/task checks in a way that we could auth with one or the other?

Reply:
I don't think it would be difficult to make a third handler that `||`s them.
I suggest keeping separate routes to enable granting least privilege for instances where only one use is anticipated. For instance, only the git.push task will try accessing author information.